r/R/differential_expression.R

```diff
@@ -50,17 +49,23 @@ runDE <- function(req, data) {
   # Check if the gene_symbol does not appear in annotation. In that case the NA value will be changed to ENSEMBL ID
   result$gene_names[is.na(result$gene_names)] <- result$Gene[is.na(result$gene_names)]
 
-  message("Paginating results:  ", str(result))
-  order_by <- pagination$orderBy
-  order_decreasing <- pagination$orderDirection == "DESC"
-  offset <- pagination$offset
-  limit <- pagination$limit
+  if("pagination" %in% names(req$body)){
+    message("Paginating results:  ", str(result))
+    pagination <- req$body$pagination
+    order_by <- pagination$orderBy
+    order_decreasing <- pagination$orderDirection == "DESC"
+    offset <- pagination$offset
+    limit <- pagination$limit
 
-  filter <- NULL
-  if ("geneNamesFilter" %in% names(req$body)) {
-    filter <- req$body$geneNamesFilter
-  }
+    filter <- NULL
+    if ("geneNamesFilter" %in% names(req$body)) {
+      filter <- req$body$geneNamesFilter
+    }
+
+    result <- handle_pagination(result, offset, limit, order_by, order_decreasing, filter)
 
-  result <- handle_pagination(result, offset, limit, order_by, order_decreasing, filter)
+  }else{
+    result <- list(gene_results = result,full_count=nrow(result))
+  }
```

Comment:
You can simplify that code if you just swap the logic if the if statements:

```
if("pagination" not %in% names(req$body)){
   message("There is no pagination, returning some other awesome results")
   return(list(gene_results = result,full_count=nrow(result)))
}

message("Paginating results:  ", str(result))
pagination <- req$body$pagination
order_by <- pagination$orderBy
order_decreasing <- pagination$orderDirection == "DESC"
offset <- pagination$offset
limit <- pagination$limit

filter <- NULL
if ("geneNamesFilter" %in% names(req$body)) {
  filter <- req$body$geneNamesFilter
}

return(handle_pagination(result, offset, limit, order_by, order_decreasing, filter))
```

Reply:
done